ui/src/views/Hooks/ListHooks/index.jsx

```diff
@@ -37,23 +37,29 @@ export default class ListHooks extends Component {
 
   handleHookSearchSubmit = hookSearch => {
     const query = parse(window.location.search.slice(1));
+    const { location, history } = this.props;
 
-    this.props.history.push({
-      search: stringify({
+    if (query.search !== hookSearch) {
+      const newQuery = {
         ...query,
         search: hookSearch,
-      }),
-    });
+
+      };
+      history.push({
+        search: stringify(newQuery, {addQueryPrefix: true}),
```

Comment:
I'm curious why this was required?  This causes `stringify` to add a `?` to the beginning of the query, but [history already does that](https://github.com/ReactTraining/history/blob/3f69f9e07b0a739419704cffc3b3563133281548/modules/LocationUtils.js#L18-L23).

It looks like history is documented as expecting the `?` to be present so this is not a bad change to make, but doesn't address the issue this PR is solving.

Reply:
Here if i try to use this **logic** will it would be correct to use it,

```
handleHookSearchSubmit = hookSearch => {
    const query = parse(window.location.search.slice(1));
    const {
        location,
        history
    } = this.props;

    if (query.search !== hookSearch) {
        const newQuery = {
            ...query,
            search: hookSearch
        }
        else {
            location.search = '?' + location.search;
        }

    }
}
```

Reply:
I think it would be correct to omit the change entirely in this PR, and focus on fixing the issue in #3044.